examples/simple-genai-server/main.go

```diff
@@ -0,0 +1,288 @@
+// Copyright 2024 Google LLC All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"net"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"agones.dev/agones/pkg/util/signals"
+	sdk "agones.dev/agones/sdks/go"
+)
+
+// main starts a UDP or TCP server
+func main() {
+	sigCtx, _ := signals.NewSigKillContext()
+
+	port := flag.String("port", "7654", "The port to listen to traffic on")
+	udp := flag.Bool("udp", true, "Server will listen on UDP")
+	tcp := flag.Bool("tcp", false, "Server will listen on TCP")
+	endpoint := flag.String("endpoint", "", "The full base URL to send API requests to")
+
+	flag.Parse()
+	if ep := os.Getenv("PORT"); ep != "" {
+		port = &ep
+	}
+	if eudp := os.Getenv("UDP"); eudp != "" {
+		u := strings.ToUpper(eudp) == "TRUE"
+		udp = &u
+	}
+	if etcp := os.Getenv("TCP"); etcp != "" {
+		t := strings.ToUpper(etcp) == "TRUE"
+		tcp = &t
+	}
+	if endp := os.Getenv("ENDPOINT"); endp != "" {
+		endpoint = &endp
+	}
+
+	log.Print("Creating SDK instance")
+	s, err := sdk.NewSDK()
+	if err != nil {
+		log.Fatalf("Could not connect to sdk: %v", err)
+	}
+
+
+	log.Print("Starting Health Ping")
+	// TODO: Is "cancel" still being used even if no method explicitly uses it?
+	ctx, cancel := context.WithCancel(context.Background())
+	go doHealth(s, ctx)
+
+	log.Printf("Creating Client at endpoint %s", *endpoint)
+	clientConn, err := initClient(*endpoint)
+
+	if *udp {
+		go udpListener(port, s, cancel, clientConn)
+	}
+
+	if *tcp {
+		go tcpListener(port, s, cancel, clientConn)
+	}
+
+	log.Print("Marking this server as ready")
+	ready(s)
+
+	<-sigCtx.Done()
+	os.Exit(0)
+}
+
+func initClient(endpoint string) (*connection, error) {
+	// TODO: create option for a client certificate
+	client := &http.Client{}
+	conn := &connection{client: client, endpoint: endpoint}
+	return conn, nil
```

Comment:
This always returns a nil error. Unless there is a chance of producing an error in this function it can be simplified to just return the connection.